src/components/DatePickerSingle.react.js

```diff
@@ -67,8 +67,8 @@ export default class DatePickerSingle extends Component {
 
     onDateChange(date) {
         const {setProps} = this.props;
-        if (setProps && date !== null) {
-            setProps({date: date.format('YYYY-MM-DD')});
+        if (setProps) {
+            setProps({date: date ? date.format('YYYY-MM-DD') : null });
```

Comment:
This looks good to me. I'm a bit concerned about backward compatibility though, as it's been this way for as long as these components have existed and users may depend on `null` never arriving in their callbacks. @chriddyp thoughts? Do we need to make this behavior opt-in?

Reply:
OK, we discussed this internally, the consensus is we're happy to call the old behavior a bug, this PR a fix, so no need for opt-in 🎉